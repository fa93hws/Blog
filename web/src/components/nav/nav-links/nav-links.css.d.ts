// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'blueA400': string;
  'gridBaseline': string;
  'link': string;
  'list': string;
  'monoBlackA100': string;
  'monoBlackA150': string;
  'monoGreyA100': string;
  'monoGreyUltraLight': string;
}
export const cssExports: CssExports;
export default cssExports;
