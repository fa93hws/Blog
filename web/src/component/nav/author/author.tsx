import * as React from 'react';
import styles from './author.css';

export type AuthorProps = {
  name: string;
  email: string;
  avatar: {
    img: string;
    alt: string;
    linkTo: string;
  };
};

export const Author = ({ name, email, avatar }: AuthorProps) => (
  <div className={styles.container}>
    <a href={avatar.linkTo}>
      <img src={avatar.img} alt={avatar.alt} className={styles.avatarImg} />
    </a>
    <p className={styles.name}>{name}</p>
    <p className={styles.email}>{email}</p>
  </div>
);
