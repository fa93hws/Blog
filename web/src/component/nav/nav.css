.nav {
  display: flex;
  flex-direction: column;
}

.avatar {
  flex-grow: 3;
  background-color: rgb(74, 74, 221);
}

.content {
  flex-grow: 7;
  background-color: rgb(44, 35, 35);
}
