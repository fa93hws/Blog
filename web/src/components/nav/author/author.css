@value gridBaseline from '../../metrics.css';
@value colorBlueLight, monoWhite, colorPurpleLight from '../../colors.css';

@value marginLeft: calc(gridBaseline * 3);

.container {
  background-color: colorBlueLight;
  padding: calc(gridBaseline * 5) 0;
}

.avatarImg {
  width: calc(gridBaseline * 10);
  height: calc(gridBaseline * 10);
  border-radius: 50%;
  margin-left: marginLeft;
}

.name {
  white-space: nowrap;
  color: monoWhite;
  margin-top: gridBaseline;
  margin-left: marginLeft;
}

.email {
  white-space: nowrap;
  color: colorPurpleLight;
  margin-top: calc(gridBaseline * 0.5);
  margin-left: marginLeft;
}
