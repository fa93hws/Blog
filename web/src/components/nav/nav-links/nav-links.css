@value gridBaseline from '../../metrics.css';
@value monoGreyA100, monoGreyUltraLight, monoBlackA100, monoBlackA150, blueA400 from '../../colors.css';

.link {
  display: block;
  line-height: calc(gridBaseline * 6);
  padding-left: calc(gridBaseline * 8);
  height: 100%;
}

.link:hover {
  color: monoBlackA100;
  background-color: blueA400;
}

.list {
  padding-top: gridBaseline;
  border-right: 1px solid monoGreyA100;
  box-shadow: 0 0 10px 0 monoGreyA100;
  box-sizing: border-box;
  background-color: monoGreyUltraLight;
  color: monoBlackA150;
  flex-grow: 1;
  overflow-y: auto;
}
