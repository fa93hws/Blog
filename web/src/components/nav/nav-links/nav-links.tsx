import * as React from 'react';
import { Link } from 'react-router-dom';
import styles from './nav-links.css';

export type NavItem = {
  text: string;
  link: string;
};
const NavLink = ({ item: { text, link } }: { item: NavItem }) => (
  <li>
    <Link to={link} className={styles.link}>
      {text}
    </Link>
  </li>
);

export const NavLinks = ({ items }: { items: readonly NavItem[] }) => (
  <ul className={styles.list}>
    {items.map((item, idx) => (
      <NavLink item={item} key={idx} />
    ))}
  </ul>
);
