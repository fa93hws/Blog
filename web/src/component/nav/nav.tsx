import * as React from 'react';
import styles from './nav.css';
import { Author, AuthorProps } from './author/author';

export const Nav = ({ author }: { author: AuthorProps }) => (
  <nav className={styles.nav}>
    <Author name={author.name} email={author.email} avatar={author.avatar} />
    <div className={styles.content}>This is content</div>
  </nav>
);
