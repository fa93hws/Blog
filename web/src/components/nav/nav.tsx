import * as React from 'react';
import styles from './nav.css';
import { Author, AuthorProps } from './author/author';
import { SourceCodeBanner } from './banner/banner';
import { NavItem, NavLinks } from './nav-links/nav-links';

export const Nav = ({
  author,
  sourceCodeUrl,
  navItems,
}: {
  author: AuthorProps;
  sourceCodeUrl: string;
  navItems: readonly NavItem[];
}) => (
  <nav className={styles.nav}>
    <SourceCodeBanner sourceCodeUrl={sourceCodeUrl} />
    <Author name={author.name} email={author.email} avatar={author.avatar} />
    <NavLinks items={navItems} />
  </nav>
);
