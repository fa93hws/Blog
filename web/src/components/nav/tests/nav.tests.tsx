import * as React from 'react';
import { MemoryRouter } from 'react-router';
import { Nav } from '../nav';
import { AuthorProps } from '../author/author';
import { NavItem } from '../nav-links/nav-links';

describe('Global Nav', () => {
  it('should match the snapshot', () => {
    const author: AuthorProps = {
      name: 'author name',
      email: 'author email',
      avatar: {
        img: 'author avatar path',
        alt: 'author avatar alt',
        linkTo: 'author avatar link',
      },
    };
    const items: readonly NavItem[] = [
      {
        text: 'item1',
        link: 'link1',
      },
      {
        text: 'item2',
        link: 'link2',
      },
    ];
    expect(
      <MemoryRouter>
        <Nav author={author} navItems={items} sourceCodeUrl="sourceCodeUrl" />
      </MemoryRouter>,
    ).toMatchRenderedSnapshot();
  });
});
