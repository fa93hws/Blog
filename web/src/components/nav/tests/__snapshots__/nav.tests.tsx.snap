// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global Nav should match the snapshot 1`] = `
<nav
  class="nav"
>
  <div
    class="banner"
  >
    <a
      href="sourceCodeUrl"
    >
      <p
        class="textMicro"
      >
        源代码
      </p>
    </a>
  </div>
  <div
    class="container"
  >
    <a
      href="author avatar link"
    >
      <img
        alt="author avatar alt"
        class="avatarImg"
        src="author avatar path"
      />
    </a>
    <div
      class="name"
    >
      <p
        class="textMicro"
      >
        author name
      </p>
    </div>
    <div
      class="email"
    >
      <p
        class="textMicro"
      >
        author email
      </p>
    </div>
  </div>
  <ul
    class="list"
  >
    <li>
      <a
        class="link"
        href="/link1"
      >
        <p
          class="textMicro"
        >
          item1
        </p>
      </a>
    </li>
    <li>
      <a
        class="link"
        href="/link2"
      >
        <p
          class="textMicro"
        >
          item2
        </p>
      </a>
    </li>
  </ul>
</nav>
`;
